### ISSUE TITLE
clickhouse: the count of view less then actual

### ISSUE THREAD

**Issue by justwinter** [ClickHouse/ClickHouse#39602]
### Discussed in https://github.com/ClickHouse/ClickHouse/discussions/39594

<div type='discussions-op-text'>

<sup>Originally posted by **justwinter** July 26, 2022</sup>
Hi,everybody:
    I have a distributed table ,and create view based on the distributed table .
  When the count of distributed table is small,the query is normal. At this time, the count of the view is the same as that of the distribution table.

  However, when the count of the distributed table is large, for example, when the count is greater than 1000000 rows, the query is abnormal. At this time, the count of views is different from that of distributed tables, only a little more than that of local tables.

 The problem occurs when I use jdbc driver to query Clickhouse view, but when I use Clickhouse-client to query, the count of the view is normal. I have tried all versions of JDBC Driver and got the same results .

My clickhouse version is 22.1.3.7

 Have you encountered similar problems? Or do you have any good solutions? 

Here is my test SQL：
CREATE TABLE IF NOT EXISTS temp.test_table_local ON CLUSTER cluster_test
(
     id String
    ,name String 
    ,desc String
    ,w_insert_dt String
) ENGINE = ReplacingMergeTree ORDER BY id SETTINGS index_granularity = 8192;

CREATE TABLE IF NOT EXISTS temp.test_table ON CLUSTER cluster_test
(
     id String
    ,name String 
    ,desc String
    ,w_insert_dt String
) ENGINE = Distributed(cluster_test,temp,test_table_local,metroHash64(id));

create view temp.test_table_v ON CLUSTER cluster_test as 
select * from temp.test_table;

insert into temp.test_table
select concat(toString(t.uuid),toString(b.sort_num)),t.name,t.comment,now() 
from `system`.tables t 
left join (select arrayJoin( range(1,1000)) as sort_num )  b on 1=1
where t.uuid not in ('00000000-0000-0000-0000-000000000000');

select count(1),max(id) from temp.test_table;
-- 2081916	fff61eea-90d4-471d-bff6-1eea90d4471d999

select count(1),max(id) from temp.test_table_v;
-- 742035	fff61eea-90d4-471d-bff6-1eea90d4471d995

select count(1),max(id) from temp.test_table_local;
-- 692883	fff61eea-90d4-471d-bff6-1eea90d4471d995

</div>

**Comment by den-crane** [comment]
JDBC drivers use HTTP protocol. You can check the same query with curl  https://clickhouse.com/docs/en/interfaces/http/

**Comment by justwinter** [comment]
> JDBC drivers use HTTP protocol. You can check the same query with curl https://clickhouse.com/docs/en/interfaces/http/
> 
> 22.1.3.7 is out of support. try with 22.7

I tried to query the count of distributed tables and views with curl command, and the result was normal

**Comment by den-crane** [comment]
continuation https://github.com/ClickHouse/clickhouse-jdbc/issues/1013

**Comment by justwinter** [comment]
` select name, value from system.settings where changed`

I found max_result_rows/result_overflow_mode these two settings are set by dbeaver, and I can't changes it.
So I tried to upgrade dbeaver, and then query was normal.
I get the same and correct results when querying views and distributed tables.

In addition, for Java programs using JDBC connection, setting a larger value for max_result_rows in the JDBC string ,I get the correct value .

I'm very sorry for wasting your time because of this small problem. At the same time, thank you very much for your help!
